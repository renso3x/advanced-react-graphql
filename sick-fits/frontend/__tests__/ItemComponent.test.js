import ItemComponent from '../components/Item';
import { shallow } from 'enzyme';

const fakeItem = {
  id: 1,
  title: 'A cool hat',
  image: 'cat.jpg',
  largeImage: 'cat_large.jpg',
  price: '50000'
}

describe('<Item />', () => {

  it('renders the image properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const image = wrapper.find('img');
    expect(image.props().src).toBe(fakeItem.image);
    expect(image.props().alt).toBe(fakeItem.title);
  });

  it('renders the button properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const buttonList = wrapper.find('.buttonList').children();
    expect(buttonList).toHaveLength(3);
    expect(buttonList.find('Link')).toBeTruthy();
    expect(buttonList.find('AddCart')).toBeTruthy();
    expect(buttonList.find('DeleteItem')).toBeTruthy();
  });

  it('renders the title and price properly', () => {
    const wrapper = shallow(<ItemComponent item={fakeItem} />);
    const title = wrapper.find('Title a').text();
    expect(title).toBe(fakeItem.title);

    const PriceTag = wrapper.find('PriceTag');
    expect(PriceTag.children().text()).toBe('$500');
  })

})