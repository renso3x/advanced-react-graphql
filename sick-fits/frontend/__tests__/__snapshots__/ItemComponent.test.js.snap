// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item /> takes a snapshot 1`] = `
<ItemStyles__Item>
  <img
    alt="A cool hat"
    src="cat.jpg"
  />
  <Title>
    <Link
      href={
        Object {
          "pathname": "/item",
          "query": Object {
            "id": 1,
          },
        }
      }
    >
      <a>
        A cool hat
      </a>
    </Link>
  </Title>
  <PriceTag>
    $500
  </PriceTag>
  <p />
  <div
    className="buttonList"
  >
    <Link
      href={
        Object {
          "pathname": "update",
          "query": Object {
            "id": 1,
          },
        }
      }
    >
      <a>
        Edit ✏️
      </a>
    </Link>
    <AddCart
      id={1}
    />
    <DeleteItem
      id={1}
    >
      Delete this Item
    </DeleteItem>
  </div>
</ItemStyles__Item>
`;
